components/google_sheets/actions/find-row/find-row.js

```diff
@@ -0,0 +1,60 @@
+const googleSheets = require("../../google_sheets.app");
+
+module.exports = {
+  key: "google_sheets-find-row",
+  name: "Find Row",
+  description: "Find a row by a column and value",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    googleSheets,
+    drive: {
+      propDefinition: [
+        googleSheets,
+        "watchedDrive",
+      ],
+      description: "",
+    },
+    sheetId: {
+      propDefinition: [
+        googleSheets,
+        "sheetID",
+        (c) => ({
+          driveId: googleSheets.methods.getDriveId(c.drive),
+        }),
+      ],
+    },
+    sheetName: {
+      propDefinition: [
+        googleSheets,
+        "sheetName",
+        (c) => ({
+          sheetId: c.sheetId,
+        }),
+      ],
+    },
+    column: {
+      propDefinition: [
+        googleSheets,
+        "column",
+      ],
+    },
+    value: {
+      type: "string",
+      label: "Value",
+      description: "The value to search for",
+    },
+  },
+  async run() {
+    const sheets = this.googleSheets.sheets();
+
+    const colValues = (await sheets.spreadsheets.values.get({
+      spreadsheetId: this.sheetId,
+      range: `${this.sheetName}!${this.column}:${this.column}`,
+    })).data.values;
+    return colValues.map((value, index) => {
+      if (value == this.value)
+        return index + 1;
+    }).filter((value) => value);
```

Comment:
Would be better (easier to read and reason about) and more efficient to filter first, then map:

```suggestion
    return colValues.filter(value => value == this.value)
      .map((value, index) => index + 1);
```

Reply:
When I test this with filtering first, then mapping, it seems to actually change the index/row values. For instance, if I have a column C with value 'z' in rows 5 and 13, it should return [ 5, 13 ]. When I filter it first, I get [ 1, 2 ]. Is there a better way to write this where we can keep the correct row numbers?

Reply:
Good point. In that case, you can combine both operations with reduce:

```suggestion
    return colValues.reduce((values, value, index) => {
      if (value == this.value) {
        values.push(index + 1);
      }
      return values;
    });
```

Reply:
Updated